plugins/scroll-options/src/ScrollBlockDragger.js

```diff
@@ -204,11 +239,187 @@ export class ScrollBlockDragger extends Blockly.BlockDragger {
       return;
     }
 
+    // Update the autoscroll or start a new one.
     this.activeAutoScroll_ =
         this.activeAutoScroll_ || new AutoScroll(this.workspace_);
     this.activeAutoScroll_.updateProperties(overallScrollVector);
   }
 
+  /**
+   * There could be multiple candidate scrolls for each direction, such as one
+   * for block position and one for mouse position. We should first find the
+   * fastest scroll in each direction. Then, we sum those to find the overall
+   * scroll vector.
+   *
+   * For example, we may have a fast block scroll and a slow
+   * mouse scroll candidate in both the top and left directions. First, we
+   * reduce to only the fast block scroll. Then, we sum the vectors in each
+   * direction to get a resulting fast scroll in a diagonal direction to the top
+   * left.
+   * @param {!CandidateScrolls} candidateScrolls Existing lists of candidate
+   *     scrolls. Will be modified in place.
+   * @return {!Blockly.utils.Coordinate} Overall scroll vector.
+   * @protected
+   */
+  getOverallScrollVector_(candidateScrolls) {
+    let overallScrollVector = new Blockly.utils.Coordinate(0, 0);
+    for (const direction of this.scrollDirections_) {
+      const scroll = candidateScrolls[direction].reduce((fastest, current) => {
```

Comment:
nit: scroll -> fastestScroll or maybe just add a comment "// Find the fastest scroll in the given direction".